examples/render/quad_render/main.rs

```diff
@@ -197,39 +128,40 @@ fn main() {
     println!("Memory types: {:?}", context.ref_device().memory_types());
     println!("Memory heaps: {:?}", context.ref_device().memory_heaps());
 
+    let mut init_tokens = Vec::new();
     let vertex_count = QUAD.len() as u64;
-    let vertex_buffer = context.mut_device().create_buffer::<Vertex, _>(
+    let (vertex_buffer, token) = device.create_buffer::<Vertex, _>(
         &mut upload,
         gfx::buffer::VERTEX,
         vertex_count
     ).unwrap();
+    init_tokens.push(token);
 
-    context.mut_device()
-        .write_mapping(&vertex_buffer, 0..vertex_count)
+    device.write_mapping(&vertex_buffer, 0..vertex_count)
```

Comment:
I wonder if you can actually do anything with that buffer, including the mapping, before it's initialized. E.g. doesn't Vulkan require the HOST access to be set?